### ISSUE TITLE
Azure: How to specify both model and deployment?

### ISSUE THREAD

**Issue by ivantheterrible** [paul-gauthier/aider#1198]
### Issue

TLDR: With Azure OpenAI, the deployment name is a different thing to the model name. Aider presumably needs to know both.  I'm not clear how you're supposed to specify both the model and the deployment?

Longer description: 

With Azure OpenAI, the --model parameter seems to be where you specify azure *deployment* name.
For example, if I have a deployment is called TEST, I  might invoke aider thus:

aider --no-auto-commits --model azure/TEST

it will then work, but it says::

Model azure/TEST: Unknown context window size and costs, using sane defaults.

Let's say the TEST deployment is in fact a gpt-4o model. You can invoke aider like this:

aider --no-auto-commits --model azure/gpt-4o

But then it doesn't actually know the deployment name, so you get an error when making the first request:

Unexpected error: litellm.NotFoundError: AzureException NotFoundError - Error code: 404 - {'error': {'code':
'DeploymentNotFound', 'message': 'The API deployment for this resource does not exist. If you created the deployment
within the last 5 minutes, please wait a moment and try again.'}}

### Version and model info

Aider v0.53.0
Azure OpenAI gpt-4o

**Comment by paul-gauthier** [comment]
Thanks for trying aider and filing this issue. These docs may be helpful:

https://aider.chat/docs/llms/azure.html
https://aider.chat/docs/config/adv-model-settings.html

You may need to manually configure all the settings and metadata for your Azure deployment.

**Comment by paul-gauthier** [comment]
You will need a model metadata and model settings file if you want to fully configure an unknown model for aider.


See [litellm's model_prices_and_context_window.json file](https://github.com/BerriAI/litellm/blob/main/model_prices_and_context_window.json) for model metadata examples.

You can look at the `ModelSettings` class in [models.py](https://github.com/paul-gauthier/aider/blob/main/aider/models.py) file for details about all of the model setting that aider supports. That file also contains the settings for many popular models.

**Comment by paul-gauthier** [comment]
I'm going to close this issue for now, but feel free to add a comment here and I will re-open. Or feel free to file a new issue any time.
